mlflow/azureml/__init__.py

```diff
@@ -380,15 +382,41 @@ def deploy(
             _copy_file_or_tree(src=absolute_model_path, dst=model_directory_path),
         )
 
-        registered_model = AzureModel.register(
-            workspace=workspace, model_path=tmp_model_path, model_name=model_name, tags=tags
-        )
+        registered_model = None
+
+        try:
+            if model_uri.startswith("models:/"):
+                m_name = model_uri.split("/")[-2]
+                m_version = int(model_uri.split("/")[-1])
+                m_id = "{}:{}".format(m_name, m_version)
+                registered_model = AzureModel(workspace, id=m_id)
+
+                _logger.info("Found registered model in AzureML with ID '%s'", m_id)
+            elif model_uri.startswith("runs:/") and get_tracking_uri().startswith("azureml"):
+                m = mlflow_register_model(model_uri, model_name)
+                m_id = "{}:{}".format(m.name, m.version)
+                registered_model = AzureModel(workspace, id=m_id)
+
+                _logger.info(
+                    "Registered an Azure Model with name: `%s` and version: `%s`",
+                    registered_model.name,
+                    registered_model.version,
+                )
+        except Exception:  # pylint: disable=broad-except
+            _logger.info(
+                "Unable to find model in AzureML with ID '%s', will register the model.", m_id
```

Comment:
We should log the error info here so that users can send us it if they see a print that does not make sense